api/v1beta1/foundationdbrestore_types.go

```diff
@@ -93,18 +96,26 @@ type FoundationDBKeyRange struct {
 // BackupName gets the name of the backup for the source backup.
 // This will fill in a default value if the backup name in the spec is empty.
 func (restore *FoundationDBRestore) BackupName() string {
-	if restore.Spec.BlobStoreConfiguration == nil || restore.Spec.BlobStoreConfiguration.BackupName == "" {
-		return restore.ObjectMeta.Name
+
+	if restore.Spec.BlobStoreConfiguration != nil && restore.Spec.BlobStoreConfiguration.BackupName != "" {
+		return restore.Spec.BlobStoreConfiguration.BackupName
+	}
+
+	if restore.Spec.FSConfiguration != nil && restore.Spec.FSConfiguration.BackupName != "" {
+		return restore.Spec.FSConfiguration.BackupName
 	}
 
-	return restore.Spec.BlobStoreConfiguration.BackupName
+	return restore.ObjectMeta.Name
 }
 
 // BackupURL gets the destination url of the backup.
 func (restore *FoundationDBRestore) BackupURL() string {
 	if restore.Spec.BlobStoreConfiguration != nil {
 		return restore.Spec.BlobStoreConfiguration.getURL(restore.BackupName(), restore.Spec.BlobStoreConfiguration.BucketName())
 	}
+	if restore.Spec.FSConfiguration != nil {
+		return fmt.Sprintf("%s/%s", restore.Spec.FSConfiguration.URL, restore.BackupName())
```

Comment:
if this is a filesystem path, you should use
```suggestion
		return path.Join(restore.Spec.FSConfiguration.URL, restore.BackupName())
```

Reply:
Same as above ?